<!DOCTYPE html>
<html lang="en-US">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <link rel="stylesheet" href="/assets/css/main.css">
    <link rel="stylesheet" href="/assets/css/font-awesome.min.css">

    <link rel="shortcut icon" href="/favicon.ico?1">
    <!-- Begin Jekyll SEO tag v2.8.0 -->
<title>deconstructSigs (NNLS) | Jake Conway</title>
<meta name="generator" content="Jekyll v3.9.3" />
<meta property="og:title" content="deconstructSigs (NNLS)" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="One major difference between NNLS and NMF is that NNLS can work in individual samples rather than requiring an entire cohort. However, depending on the number of mutations per sample, NMF may be powered enough to provide insight on individual samples. With whole-exome sequencing data, this is likely not the case." />
<meta property="og:description" content="One major difference between NNLS and NMF is that NNLS can work in individual samples rather than requiring an entire cohort. However, depending on the number of mutations per sample, NMF may be powered enough to provide insight on individual samples. With whole-exome sequencing data, this is likely not the case." />
<link rel="canonical" href="http://localhost:4000/docs/deconstructSigs/" />
<meta property="og:url" content="http://localhost:4000/docs/deconstructSigs/" />
<meta property="og:site_name" content="Jake Conway" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2023-06-04T23:07:34+02:00" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="deconstructSigs (NNLS)" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"WebPage","dateModified":"2023-06-04T23:07:34+02:00","datePublished":"2023-06-04T23:07:34+02:00","description":"One major difference between NNLS and NMF is that NNLS can work in individual samples rather than requiring an entire cohort. However, depending on the number of mutations per sample, NMF may be powered enough to provide insight on individual samples. With whole-exome sequencing data, this is likely not the case.","headline":"deconstructSigs (NNLS)","url":"http://localhost:4000/docs/deconstructSigs/"}</script>
<!-- End Jekyll SEO tag -->


    <link rel="alternate" type="application/rss+xml" title="Jake Conway" href="http://localhost:4000/feed.xml" />
</head>


<body>

    <nav class="navbar navbar-default navbar-fixed-top">
    <div class="container navbar-container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
            <a class="navbar-brand" href="//">
                <span><img src="/assets/img/logonav.png" alt="Logo"></span> Jake Conway
            </a>
        </div>
        <div id="navbar" class="collapse navbar-collapse">
            <ul class="nav navbar-nav">
                <li ><a href="/CV/home/">CV</a></li>
                <li  class="active" ><a href="/docs/home/">Comp Bio</a></li>
                <li ><a href="/stats/home/">Stats+ML</a></li>
                <li ><a href="/blog/2017/05/03/jekyll-doc-theme/">Blog</a></li>
            </ul>
            <div class="navbar-right">
                <form class="navbar-form navbar-left">
                    <div class="form-group has-feedback">
                        <input id="search-box" type="search" class="form-control" placeholder="Search...">
                        <i class="fa fa-search form-control-feedback"></i>
                    </div>
                </form>
                <ul class="nav navbar-nav">
                    <li><a href="https://github.com/JakeConway"><i class="fa fa-github" aria-hidden="true"></i></a></li>
                </ul>
            </div>
        </div>
    </div>
</nav>


    <div class="page-content">
        <div class="wrapper">
            <div class="container">
    <div class="row">
        <div class="col-md-4">
          <div class="panel-group" id="accordion" role="tablist" aria-multiselectable="true">
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-1" aria-expanded="false" aria-controls="collapse-1">
            Getting Started
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-1" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/home/">Welcome</a>
          
            
            
            <a class="list-group-item " href="/docs/bioconductor/">Installing Bioconductor</a>
          
            
            
            <a class="list-group-item " href="/docs/commandlinetools/">Command line tools (Docker)</a>
          
            
            
            <a class="list-group-item " href="/docs/getting_data/">Getting the data</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-2" aria-expanded="false" aria-controls="collapse-2">
            Quality Control
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-2" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewQC/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/SequencingCoverage/">Depth of Sequencing Coverage</a>
          
            
            
            <a class="list-group-item " href="/docs/CrossSampleContamination/">Cross Sample Contamination</a>
          
            
            
            <a class="list-group-item " href="/docs/TumorInNormal/">Tumor-in-normal</a>
          
            
            
            <a class="list-group-item " href="/docs/purity/">Tumor Purity</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-3" aria-expanded="false" aria-controls="collapse-3">
            Somatic Mutation Calling
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-3" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewSomaticMutations/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/MuTect/">MuTect</a>
          
            
            
            <a class="list-group-item " href="/docs/Strelka/">Strelka</a>
          
            
            
            <a class="list-group-item " href="/docs/MuTect2/">MuTect2</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-4" aria-expanded="false" aria-controls="collapse-4">
            Allelic Copy Number Calling
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-4" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewAllelicCNA/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/FACETS/">FACETS</a>
          
            
            
            <a class="list-group-item " href="/docs/GATKACNV/">GATK ACNV</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-5" aria-expanded="false" aria-controls="collapse-5">
            Mutational Signatures
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse" id="collapse-5" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewMutationalSignatures/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/SomaticSignatures/">SomaticSignatures (NMF)</a>
          
            
            
            <a class="list-group-item active" href="/docs/deconstructSigs/">deconstructSigs (NNLS)</a>
          
            
            
            <a class="list-group-item " href="/docs/SigMa/">SigMa (Likelihood)</a>
          
            
            
            <a class="list-group-item " href="/docs/scarHRD/">scarHRD (Homologous Recombination Deficiency)</a>
          
            
            
            <a class="list-group-item " href="/docs/MSIsensor/">MSIsensor (Microsatellite Instability)</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-6" aria-expanded="false" aria-controls="collapse-6">
            Mutational Significance Analysis
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-6" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewSMG/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/MutSigCV/">MutSigCV</a>
          
            
            
            <a class="list-group-item " href="/docs/OncodriveFML/">OncodriveFML</a>
          
            
            
            <a class="list-group-item " href="/docs/MutPanning/">MutPanning</a>
          
            
            
            <a class="list-group-item " href="/docs/BrownsMethod/">Brown's Method (for combining p-values)</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-7" aria-expanded="false" aria-controls="collapse-7">
            Tumor Phylogenetics
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-7" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewPhylo/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/CCF/">Calculate Mutation Cancer Cell Fractions (CCFs)</a>
          
            
            
            <a class="list-group-item " href="/docs/PhylogicNDT/">PhylogicNDT</a>
          
            
            
            <a class="list-group-item " href="/docs/PyClone/">PyClone</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-8" aria-expanded="false" aria-controls="collapse-8">
            Gene Set Analysis
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-8" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewGeneSets/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/GSEA/">GSEA</a>
          
            
            
            <a class="list-group-item " href="/docs/ssGSEA/">ssGSEA</a>
          
            
            
            <a class="list-group-item " href="/docs/PathwayOverrepresentation/">Pathway Overrepresentation</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-9" aria-expanded="false" aria-controls="collapse-9">
            Differential Expression
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-9" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewDifferentialExpression/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/edgeR/">edgeR</a>
          
            
            
            <a class="list-group-item " href="/docs/DESeq2/">DESeq2</a>
          
            
            
            <a class="list-group-item " href="/docs/IHW/">Independent Hypothesis Weighting (IHW)</a>
          
        </div>
      </div>
    </div>
  
</div>

        </div>

        <div class="col-md-8">
            <h1>deconstructSigs (NNLS)</h1>
            <div id="markdown-content-container"><p>One major difference between NNLS and NMF is that NNLS can work in individual samples rather than requiring an entire cohort. However, depending on the number of mutations per sample, NMF may be powered enough to provide insight on individual samples. With whole-exome sequencing data, this is likely not the case.</p>

<p>deconstructSigs is an R package that can be downloaded through CRAN.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">deconstructSigs</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">plyr</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>In whole-exome sequencing data, we mostly care about SNV mutational signatures. Even if we were powered enough for indel or dinucleotide mutational signatures (i.e. enough indel or dinucleotide mutations), we would still run each separately. This is because the reference signatures (relative frequency of mutation in trinucleotide context for each mutational process) were determined for SNVs, indels and DNPs separately.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># only SNVs should be used for determining SNV mutational signatures</span><span class="w">
</span><span class="n">snp.maf</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">maf</span><span class="p">[</span><span class="n">which</span><span class="p">(</span><span class="n">maf</span><span class="o">$</span><span class="n">Variant_Type</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"SNP"</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="w">
</span></code></pre></div></div>

<p>deconstructSigs (NNLS) requires the same information as SomaticSignatures (NMF).</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">deconstructSigs.df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">snp.maf</span><span class="p">[</span><span class="w"> </span><span class="p">,</span><span class="nf">c</span><span class="p">(</span><span class="s2">"Tumor_Sample_Barcode"</span><span class="p">,</span><span class="w">
                                  </span><span class="s2">"Chromosome"</span><span class="p">,</span><span class="w">
                                  </span><span class="s2">"Start_position"</span><span class="p">,</span><span class="w">
                                  </span><span class="s2">"Reference_Allele"</span><span class="p">,</span><span class="w">
                                  </span><span class="s2">"Tumor_Seq_Allele2"</span><span class="p">)]</span><span class="w">
</span><span class="n">deconstructSigs.df</span><span class="o">$</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s2">"chr"</span><span class="p">,</span><span class="w"> </span><span class="n">deconstructSigs.df</span><span class="o">$</span><span class="n">Chromosome</span><span class="p">)</span><span class="w">

</span><span class="c1"># requires chrX and chrY to be encoded as chr23 and chr24 respectively (to match reference)</span><span class="w">
</span><span class="n">deconstructSigs.df</span><span class="p">[</span><span class="n">which</span><span class="p">(</span><span class="n">deconstructSigs.df</span><span class="o">$</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"chrX"</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="o">$</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"chr23"</span><span class="w">
</span><span class="n">deconstructSigs.df</span><span class="p">[</span><span class="n">which</span><span class="p">(</span><span class="n">deconstructSigs.df</span><span class="o">$</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"chrY"</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="o">$</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"chr24"</span><span class="w">
</span><span class="n">deconstructSigs.df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">deconstructSigs.df</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>Just like the NMF based method, the required input is the frequency of mutations in each trinucleotide context. However, this time it is the frequency in each sample rather than the frequency across the cohort.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># deconstructSigs signatures</span><span class="w">
</span><span class="n">sigs.input</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">deconstructSigs</span><span class="o">::</span><span class="n">mut.to.sigs.input</span><span class="p">(</span><span class="n">deconstructSigs.df</span><span class="p">,</span><span class="w">
                                                 </span><span class="n">sample.id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Tumor_Sample_Barcode"</span><span class="p">,</span><span class="w"> 
                                                 </span><span class="n">chr</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Chromosome"</span><span class="p">,</span><span class="w"> 
                                                 </span><span class="n">pos</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Start_position"</span><span class="p">,</span><span class="w"> 
                                                 </span><span class="n">ref</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Reference_Allele"</span><span class="p">,</span><span class="w"> 
                                                 </span><span class="n">alt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Tumor_Seq_Allele2"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>Since NNLS works at the individual sample level, <strong>whichSignatures</strong> only runs on one sample at a time. Thus, we use the <strong>lapply</strong> function to loop through the trinucleotide context frequencies of each sample and determine both 1) which mutational signatures are active, and 2) the reltive contribution of each active mutational signature to the mutational spectrum of the tumor.</p>

<p><strong>deconstructSigs</strong> recommends a relative signature contribution cutoff of 0.06 (6%), which through the authors in-silico analysis, led to a FDR of 1%. As a result, if there are any signatures that have a relative contribution of less than 0.06 via the NNLS model, <strong>deconstructSigs</strong> will set the relative contribution to 0, causing the sum of the relative contributions to sum to less than 1 (100%). Thus, we also havev to rescale the relative contributions of each signature such that the sum of the relative signature contributions equals 1 (100%).</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">sig.contrib.list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">row.names</span><span class="p">(</span><span class="n">sigs.input</span><span class="p">),</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
  </span><span class="n">row.df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">deconstructSigs</span><span class="o">::</span><span class="n">whichSignatures</span><span class="p">(</span><span class="n">sigs.input</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">contexts.needed</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> 
                                             </span><span class="n">signatures.ref</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">signatures.cosmic</span><span class="p">)</span><span class="o">$</span><span class="n">weights</span><span class="w">
  </span><span class="n">row.sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rowSums</span><span class="p">(</span><span class="n">row.df</span><span class="p">)</span><span class="w">
  </span><span class="n">row.df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">row.df</span><span class="o">/</span><span class="n">row.sum</span><span class="w">
  </span><span class="n">rownames</span><span class="p">(</span><span class="n">row.df</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">x</span><span class="w">
  </span><span class="nf">return</span><span class="p">(</span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">row.df</span><span class="p">))</span><span class="w">
</span><span class="p">})</span><span class="w">
</span></code></pre></div></div>

<p>After determining the relative contribution of each mutational signature in each tumor, we can merge the results into one data.frame for easier downstream analysis.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">deconstructSigs.results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plyr</span><span class="o">::</span><span class="n">rbind.fill</span><span class="p">(</span><span class="n">sig.contrib.list</span><span class="p">)</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">deconstructSigs.results</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">sigs.input</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>To provide a comparisons to the NMF method (SomaticSignatures), we can estimate the activity of each mutational signature across by summing the relative contributions for each mutational signature. A more appropriate way to compare the NNLS results with the NMF results would be to multiply the relative contributions of each mutational signature by their respective sampleâ€™s tumor mutational burden (TMB), and then summing the signature activities across the cohort.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">process.ranks</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">colSums</span><span class="p">(</span><span class="n">deconstructSigs.results</span><span class="p">)</span><span class="w">
</span><span class="n">process.ranks</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sort</span><span class="p">(</span><span class="n">process.ranks</span><span class="p">,</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
</span><span class="n">process.ranks</span><span class="w">
</span></code></pre></div></div>
</div>
            <div style="clear:both;">
              <p class="text-center">
                <br />
                <a target="_blank" href="https://github.com/JakeConway/blob/gh-pages/_docs/Mutational Signatures/deconstructSigs.md" class="btn btn-default" role="button">
                  <i class="fa fa-pencil fa-lg"></i> Improve this page
                </a>
              </p>
            </div>
            <hr>
            





  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  
    <ul class="pager">
      
        
        
        <li class="previous">
          <a href="/docs/SomaticSignatures/">
            <span aria-hidden="true">&larr;</span> Previous
          </a>
        </li>
      

      
        
        
        <li class="next">
          <a href="/docs/SigMa/">
            Next <span aria-hidden="true">&rarr;</span>
          </a>
        </li>
      
      </ul>
    <div class="clear"></div>
    

        </div>

    </div>
</div>

        </div>
    </div>

    <footer class="footer">
    <div class="container">

        <p class="text-center">
            Jake Conway 2023 |
            Powered by <a href="https://github.com/aksakalli/jekyll-doc-theme">Jekyll Doc Theme</a>
        </p>
        <!-- <p class="text-muted">Place sticky footer content here.</p> -->
    </div>
</footer>

    <script>
  var baseurl = '/'
</script>
<script src="https://code.jquery.com/jquery-1.12.4.min.js"></script>
<script src="/assets/js/bootstrap.min.js "></script>
<script src="/assets/js/typeahead.bundle.min.js "></script>

<script src="/assets/js/main.js "></script>

</body>

</html>
