<!DOCTYPE html>
<html lang="en-US">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <link rel="stylesheet" href="/assets/css/main.css">
    <link rel="stylesheet" href="/assets/css/font-awesome.min.css">

    <link rel="shortcut icon" href="/favicon.ico?1">
    <!-- Begin Jekyll SEO tag v2.8.0 -->
<title>SomaticSignatures (NMF) | Jake Conway</title>
<meta name="generator" content="Jekyll v3.9.3" />
<meta property="og:title" content="SomaticSignatures (NMF)" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Load the data.table R package for reading in files and the pheatmap R package for plotting the output from NMF analysis. # CRAN packages library(data.table) library(pheatmap)" />
<meta property="og:description" content="Load the data.table R package for reading in files and the pheatmap R package for plotting the output from NMF analysis. # CRAN packages library(data.table) library(pheatmap)" />
<link rel="canonical" href="http://localhost:4000/docs/SomaticSignatures/" />
<meta property="og:url" content="http://localhost:4000/docs/SomaticSignatures/" />
<meta property="og:site_name" content="Jake Conway" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2023-04-29T04:12:54+02:00" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="SomaticSignatures (NMF)" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"WebPage","dateModified":"2023-04-29T04:12:54+02:00","datePublished":"2023-04-29T04:12:54+02:00","description":"Load the data.table R package for reading in files and the pheatmap R package for plotting the output from NMF analysis. # CRAN packages library(data.table) library(pheatmap)","headline":"SomaticSignatures (NMF)","url":"http://localhost:4000/docs/SomaticSignatures/"}</script>
<!-- End Jekyll SEO tag -->


    <link rel="alternate" type="application/rss+xml" title="Jake Conway" href="http://localhost:4000/feed.xml" />
</head>


<body>

    <nav class="navbar navbar-default navbar-fixed-top">
    <div class="container navbar-container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
            <a class="navbar-brand" href="//">
                <span><img src="/assets/img/logonav.png" alt="Logo"></span> Jake Conway
            </a>
        </div>
        <div id="navbar" class="collapse navbar-collapse">
            <ul class="nav navbar-nav">
                <li ><a href="/CV/home/">CV</a></li>
                <li  class="active" ><a href="/docs/home/">Comp Bio</a></li>
                <li ><a href="/stats/home/">Stats+ML</a></li>
                <li ><a href="/blog/2017/05/03/jekyll-doc-theme/">Blog</a></li>
            </ul>
            <div class="navbar-right">
                <form class="navbar-form navbar-left">
                    <div class="form-group has-feedback">
                        <input id="search-box" type="search" class="form-control" placeholder="Search...">
                        <i class="fa fa-search form-control-feedback"></i>
                    </div>
                </form>
                <ul class="nav navbar-nav">
                    <li><a href="https://github.com/JakeConway"><i class="fa fa-github" aria-hidden="true"></i></a></li>
                </ul>
            </div>
        </div>
    </div>
</nav>


    <div class="page-content">
        <div class="wrapper">
            <div class="container">
    <div class="row">
        <div class="col-md-4">
          <div class="panel-group" id="accordion" role="tablist" aria-multiselectable="true">
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-1" aria-expanded="false" aria-controls="collapse-1">
            Getting Started
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-1" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/home/">Welcome</a>
          
            
            
            <a class="list-group-item " href="/docs/bioconductor/">Installing Bioconductor</a>
          
            
            
            <a class="list-group-item " href="/docs/commandlinetools/">Command line tools (Docker)</a>
          
            
            
            <a class="list-group-item " href="/docs/getting_data/">Getting the data</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-2" aria-expanded="false" aria-controls="collapse-2">
            Quality Control
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-2" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewQC/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/SequencingCoverage/">Depth of Sequencing Coverage</a>
          
            
            
            <a class="list-group-item " href="/docs/CrossSampleContamination/">Cross Sample Contamination</a>
          
            
            
            <a class="list-group-item " href="/docs/TumorInNormal/">Tumor-in-normal</a>
          
            
            
            <a class="list-group-item " href="/docs/purity/">Tumor Purity</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-3" aria-expanded="false" aria-controls="collapse-3">
            Somatic Mutation Calling
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-3" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewSomaticMutations/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/MuTect/">MuTect</a>
          
            
            
            <a class="list-group-item " href="/docs/Strelka/">Strelka</a>
          
            
            
            <a class="list-group-item " href="/docs/MuTect2/">MuTect2</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-4" aria-expanded="false" aria-controls="collapse-4">
            Allelic Copy Number Calling
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-4" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewAllelicCNA/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/FACETS/">FACETS</a>
          
            
            
            <a class="list-group-item " href="/docs/GATKACNV/">GATK ACNV</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-5" aria-expanded="false" aria-controls="collapse-5">
            Mutational Signatures
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse" id="collapse-5" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewMutationalSignatures/">Overview</a>
          
            
            
            <a class="list-group-item active" href="/docs/SomaticSignatures/">SomaticSignatures (NMF)</a>
          
            
            
            <a class="list-group-item " href="/docs/deconstructSigs/">deconstructSigs (NNLS)</a>
          
            
            
            <a class="list-group-item " href="/docs/SigMa/">SigMa (Likelihood)</a>
          
            
            
            <a class="list-group-item " href="/docs/scarHRD/">scarHRD (Homologous Recombination Deficiency)</a>
          
            
            
            <a class="list-group-item " href="/docs/MSIsensor/">MSIsensor (Microsatellite Instability)</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-6" aria-expanded="false" aria-controls="collapse-6">
            Mutational Significance Analysis
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-6" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewSMG/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/MutSigCV/">MutSigCV</a>
          
            
            
            <a class="list-group-item " href="/docs/OncodriveFML/">OncodriveFML</a>
          
            
            
            <a class="list-group-item " href="/docs/MutPanning/">MutPanning</a>
          
            
            
            <a class="list-group-item " href="/docs/BrownsMethod/">Brown's Method (for combining p-values)</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-7" aria-expanded="false" aria-controls="collapse-7">
            Tumor Phylogenetics
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-7" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewPhylo/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/CCF/">Calculate Mutation Cancer Cell Fractions (CCFs)</a>
          
            
            
            <a class="list-group-item " href="/docs/PhylogicNDT/">PhylogicNDT</a>
          
            
            
            <a class="list-group-item " href="/docs/PyClone/">PyClone</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-8" aria-expanded="false" aria-controls="collapse-8">
            Gene Set Analysis
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-8" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewGeneSets/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/GSEA/">GSEA</a>
          
            
            
            <a class="list-group-item " href="/docs/ssGSEA/">ssGSEA</a>
          
            
            
            <a class="list-group-item " href="/docs/PathwayOverrepresentation/">Pathway Overrepresentation</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-9" aria-expanded="false" aria-controls="collapse-9">
            Differential Expression
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-9" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewDifferentialExpression/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/edgeR/">edgeR</a>
          
            
            
            <a class="list-group-item " href="/docs/DESeq2/">DESeq2</a>
          
            
            
            <a class="list-group-item " href="/docs/IHW/">Independent Hypothesis Weighting (IHW)</a>
          
        </div>
      </div>
    </div>
  
</div>

        </div>

        <div class="col-md-8">
            <h1>SomaticSignatures (NMF)</h1>
            <div id="markdown-content-container"><p>Load the <code class="language-plaintext highlighter-rouge">data.table</code> R package for reading in files and the <code class="language-plaintext highlighter-rouge">pheatmap</code> R package for plotting the output from NMF analysis.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># CRAN packages</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">pheatmap</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>Load the Bioconductor packages <code class="language-plaintext highlighter-rouge">TCGAbiolinks</code> to obtain the data, <code class="language-plaintext highlighter-rouge">BSgenome.Hsapiens.1000genomes.hs37d5</code> to serve as a genome reference for mutation calls, <code class="language-plaintext highlighter-rouge">maftools</code> to obtain the 96 trinucleotide signature references, <code class="language-plaintext highlighter-rouge">SomaticSignatures</code> to perform NMF, <code class="language-plaintext highlighter-rouge">SomaticCancerAlterations</code> to convert the MAF file data to a GRanges object, and <code class="language-plaintext highlighter-rouge">VariantAnnotation</code> to convert GRange objects into VRange objects. In short, GRange objects are R objects for storing genomic location information, and VRange objects extend GRange objects to include variant call information (e.g. reference allele, alternate allele, reference allele counts, alternate allele counts, etc.).</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Bioconductor packages</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">TCGAbiolinks</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">BSgenome.Hsapiens.1000genomes.hs37d5</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">maftools</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">SomaticSignatures</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">SomaticCancerAlterations</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">VariantAnnotation</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>Download and generate the MAF for the TCGA-SKCM cohort.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">query</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">GDCquery</span><span class="p">(</span><span class="n">project</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"TCGA-OV"</span><span class="p">,</span><span class="w"> 
                  </span><span class="n">data.category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Simple nucleotide variation"</span><span class="p">,</span><span class="w"> 
                  </span><span class="n">data.type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Simple somatic mutation"</span><span class="p">,</span><span class="w">
                  </span><span class="n">access</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"open"</span><span class="p">,</span><span class="w">
                  </span><span class="n">legacy</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">

</span><span class="n">GDCdownload</span><span class="p">(</span><span class="n">query</span><span class="p">)</span><span class="w">
</span><span class="n">maf</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">GDCprepare</span><span class="p">(</span><span class="n">query</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>While indel and double nucleotide polymorphism (DNP) mutational signatures exist, the most common type of mutational signature analyses performed is SNV mutational signatures. Each type of mutational signatures (e.g. SNV, indel, DNP) has a different set of reference signatures with different etiologies. For a comprehensive list of mutational signatures please see: <a href="https://cancer.sanger.ac.uk/cosmic/signatures">https://cancer.sanger.ac.uk/cosmic/signatures</a>. The three sets of mutational signatures in red (SBS, DBS, and ID) are the <em>new</em> reference sets of signatures for SNVs, DNPs, and indels <strong>derived and validated in whole-genome sequencing (WGS) data</strong>, but not whole-exome sequencing (WES) data. <strong>Since we are using WES mutation calls, we will be using the reference set of SNV mutational signatures referred to as <a href="https://cancer.sanger.ac.uk/cosmic/signatures_v2.tt">mutational signatures version 2</a>, which were derivied from WES and validated in both WGS and WES.</strong></p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># only SNPs should be used for determining SNV mutational signatures</span><span class="w">
</span><span class="n">snp.maf</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">maf</span><span class="p">[</span><span class="n">which</span><span class="p">(</span><span class="n">maf</span><span class="o">$</span><span class="n">Variant_Type</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"SNP"</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="w">


</span><span class="c1"># Patient_ID and Sample_ID are required column names</span><span class="w">
</span><span class="n">snp.maf</span><span class="o">$</span><span class="n">Patient_ID</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">snp.maf</span><span class="o">$</span><span class="n">Tumor_Sample_Barcode</span><span class="w">
</span><span class="n">snp.maf</span><span class="o">$</span><span class="n">Sample_ID</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">snp.maf</span><span class="o">$</span><span class="n">Tumor_Sample_Barcode</span><span class="w">

</span><span class="c1"># convert MAF to GRange object</span><span class="w">
</span><span class="n">GRange</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">SomaticCancerAlterations</span><span class="o">:::</span><span class="n">.maf2gr</span><span class="p">(</span><span class="n">snp.maf</span><span class="p">,</span><span class="w"> </span><span class="n">seq_info</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">)</span><span class="w">

</span><span class="c1"># generate VRanges object from GRanges object</span><span class="w">
</span><span class="n">vr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">VariantAnnotation</span><span class="o">::</span><span class="n">VRanges</span><span class="p">(</span><span class="w">
  </span><span class="n">seqnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">seqnames</span><span class="p">(</span><span class="n">GRange</span><span class="p">),</span><span class="w">
  </span><span class="n">ranges</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ranges</span><span class="p">(</span><span class="n">GRange</span><span class="p">),</span><span class="w">
  </span><span class="n">ref</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GRange</span><span class="o">$</span><span class="n">Reference_Allele</span><span class="p">,</span><span class="w">
  </span><span class="n">alt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GRange</span><span class="o">$</span><span class="n">Tumor_Seq_Allele2</span><span class="p">,</span><span class="w">
  </span><span class="n">sampleNames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GRange</span><span class="o">$</span><span class="n">Sample_ID</span><span class="p">,</span><span class="w">
  </span><span class="n">seqinfo</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">seqinfo</span><span class="p">(</span><span class="n">GRange</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>After converting the SNV mutational calls to a VRanges object, determine the trinucleotide sequence context for each variant using the genome reference and subsequently normalize the frequency of mutations in each trinucleotide conext. The normalization is performed to adjust for the tumor mutational burden, and so that the data is in the same format as the reference set of signatures. If you go back and look at the <a href="https://cancer.sanger.ac.uk/cosmic/signatures_v2.tt">reference set of signatures on the COSMIC website</a>, you will notice the y-axis is in terms of probability (i.e. frequency).</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># generate counts for number of mutations in each of the 96 trinucleotide contexts</span><span class="w">
</span><span class="n">motifs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">SomaticSignatures</span><span class="o">::</span><span class="n">mutationContext</span><span class="p">(</span><span class="n">vr</span><span class="p">,</span><span class="w"> </span><span class="n">BSgenome.Hsapiens.1000genomes.hs37d5</span><span class="p">)</span><span class="w">

</span><span class="c1"># normalize the counts by frequency</span><span class="w">
</span><span class="n">mm</span><span class="w"> </span><span class="o">=</span><span class="w">  </span><span class="n">SomaticSignatures</span><span class="o">::</span><span class="n">motifMatrix</span><span class="p">(</span><span class="n">motifs</span><span class="p">,</span><span class="w"> </span><span class="n">normalize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>Below we will run NMF on the normalize trinucleotide counts. Here we’re running NMF to decompose the mutations into between 2 to 8 mutational signatures and performing 5 replicates for each (i.e. 5 replicates when decomposing into 2 signatures, 5 replicates when decomposing into 3 signatures, etc.). The reason we perform replicates is because noise in the mutation calls can affect the stability of factorization and convergence of the algorithm, which is reflected in the variation between the residual sum of squares (RSS), or “fit”, of the decomposition. One of the simplest ways to choose the optimal number of signatures is to choose the number of signatures such that increasing the number of signatures by 1 doesn’t result in a drastically lower RSS.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Run NMF for between 2 to 8 active signatures</span><span class="w">
</span><span class="n">n.sigs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="o">:</span><span class="m">8</span><span class="w">

</span><span class="c1"># Determine residual sum of squares and variance explained</span><span class="w">
</span><span class="n">nmf</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">SomaticSignatures</span><span class="o">::</span><span class="n">assessNumberSignatures</span><span class="p">(</span><span class="n">mm</span><span class="p">,</span><span class="w"> </span><span class="n">n.sigs</span><span class="p">,</span><span class="w"> </span><span class="n">nReplicates</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w">

</span><span class="c1"># plot residual sum of squares and variance explained</span><span class="w">
</span><span class="n">SomaticSignatures</span><span class="o">::</span><span class="n">plotNumberSignatures</span><span class="p">(</span><span class="n">nmf</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>Based on the residual RSS plot, 7 signatures appears to be the optimal number of signatures for NMF decomposition.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># based on plots, 7 signatures seems optimal</span><span class="w">
</span><span class="n">n.sigs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">7</span><span class="w">

</span><span class="c1"># perform NMF decomposition with 7 signatures </span><span class="w">
</span><span class="n">sigs.nmf</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">SomaticSignatures</span><span class="o">::</span><span class="n">identifySignatures</span><span class="p">(</span><span class="n">mm</span><span class="p">,</span><span class="w"> </span><span class="n">n.sigs</span><span class="p">,</span><span class="w"> 
                                                 </span><span class="n">SomaticSignatures</span><span class="o">::</span><span class="n">nmfDecomposition</span><span class="p">,</span><span class="w"> </span><span class="s2">"brunet"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>The result of <code class="language-plaintext highlighter-rouge">SomaticSignatures::identifySignatures</code> provides us with 2 important matrices:</p>
<ol>
  <li>An S (signatures) by N (samples) matrix, which can be interpreted as the relative contribution of each mutational signature to each samples mutational spectrum.</li>
  <li>A 96 by S (signatures) matrix, which can be interpreted as the frequency of mutations in each 96 trinucleotide bin for each signature.<br />
Now that we have this information, we want to figure out which mutational signatures each of our S signatures correspond to. To do this we will perform cosine similarity between our 96 by S matrix, and the set of reference signatures.</li>
</ol>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># extract signatures data.frame</span><span class="w">
</span><span class="n">w</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sigs.nmf</span><span class="o">@</span><span class="n">signatures</span><span class="w">

</span><span class="c1"># read in mutational signatures reference file (COSMIC V2) and format</span><span class="w">
</span><span class="n">sigs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data.table</span><span class="o">::</span><span class="n">fread</span><span class="p">(</span><span class="n">input</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">system.file</span><span class="p">(</span><span class="s1">'extdata'</span><span class="p">,</span><span class="w"> </span><span class="s1">'signatures.txt'</span><span class="p">,</span><span class="w"> </span><span class="n">package</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'maftools'</span><span class="p">),</span><span class="w">
                         </span><span class="n">stringsAsFactors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">data.table</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="n">sigs</span><span class="p">)</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gsub</span><span class="p">(</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">' '</span><span class="p">,</span><span class="w"> </span><span class="n">replacement</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'_'</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colnames</span><span class="p">(</span><span class="n">sigs</span><span class="p">))</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">sigs</span><span class="p">)</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sigs</span><span class="o">$</span><span class="n">Somatic_Mutation_Type</span><span class="w">

</span><span class="c1"># restrict to just the normalized mutation counts</span><span class="w">
</span><span class="n">sigs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sigs</span><span class="p">[,</span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">)]</span><span class="w">

</span><span class="c1"># rename rows ofsignatures data.frame to be consistent with reference</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">w</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">sigs</span><span class="p">)</span><span class="w">

</span><span class="c1"># sanity check</span><span class="w">
</span><span class="n">sigs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sigs</span><span class="p">[</span><span class="n">rownames</span><span class="p">(</span><span class="n">w</span><span class="p">),]</span><span class="w">

</span><span class="c1"># perform cosine similarity between NMF results and reference signatures</span><span class="w">
</span><span class="n">coSineMat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">()</span><span class="w">
  </span><span class="k">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">ncol</span><span class="p">(</span><span class="n">w</span><span class="p">)){</span><span class="w">
  </span><span class="n">sig</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">w</span><span class="p">[,</span><span class="n">i</span><span class="p">]</span><span class="w">
  </span><span class="n">coSineMat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rbind</span><span class="p">(</span><span class="n">coSineMat</span><span class="p">,</span><span class="w"> </span><span class="n">apply</span><span class="p">(</span><span class="n">sigs</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span><span class="w">
    </span><span class="nf">round</span><span class="p">(</span><span class="n">crossprod</span><span class="p">(</span><span class="n">sig</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="p">)</span><span class="o">/</span><span class="nf">sqrt</span><span class="p">(</span><span class="n">crossprod</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">crossprod</span><span class="p">(</span><span class="n">sig</span><span class="p">)),</span><span class="w"> </span><span class="n">digits</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">)</span><span class="w">
  </span><span class="p">}))</span><span class="w">
</span><span class="p">}</span><span class="w">

  </span><span class="n">rownames</span><span class="p">(</span><span class="n">coSineMat</span><span class="p">)</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colnames</span><span class="p">(</span><span class="n">w</span><span class="p">)</span><span class="w">

</span><span class="c1"># determine which known signature has the highest cosine similarity with each NMF signature</span><span class="w">
</span><span class="k">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">nrow</span><span class="p">(</span><span class="n">coSineMat</span><span class="p">)){</span><span class="w">
  </span><span class="n">print</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">which</span><span class="p">(</span><span class="n">coSineMat</span><span class="p">[</span><span class="n">i</span><span class="p">,]</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">coSineMat</span><span class="p">[</span><span class="n">i</span><span class="p">,]))))</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>
<p>Now that we know which reference signatures each of our decomposed signatures represents, we can plot the data to show the results! One thing you should notice is that some signature references are very similar to one another, and therefore some of our decomposed signatures may have high cosine similarity to multiple signatures.</p>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># plot the cosine simarity between each NMF signature and each known signature</span><span class="w">
</span><span class="n">pheatmap</span><span class="o">::</span><span class="n">pheatmap</span><span class="p">(</span><span class="n">mat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">coSineMat</span><span class="p">,</span><span class="w"> </span><span class="n">cluster_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
                   </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cosine similarity against validated signatures"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</div>
            <div style="clear:both;">
              <p class="text-center">
                <br />
                <a target="_blank" href="https://github.com/JakeConway/blob/gh-pages/_docs/Mutational Signatures/SomaticSignatures.md" class="btn btn-default" role="button">
                  <i class="fa fa-pencil fa-lg"></i> Improve this page
                </a>
              </p>
            </div>
            <hr>
            





  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  
    <ul class="pager">
      
        
        
        <li class="previous">
          <a href="/docs/OverviewMutationalSignatures/">
            <span aria-hidden="true">&larr;</span> Previous
          </a>
        </li>
      

      
        
        
        <li class="next">
          <a href="/docs/deconstructSigs/">
            Next <span aria-hidden="true">&rarr;</span>
          </a>
        </li>
      
      </ul>
    <div class="clear"></div>
    

        </div>

    </div>
</div>

        </div>
    </div>

    <footer class="footer">
    <div class="container">

        <p class="text-center">
            Jake Conway 2023 |
            Powered by <a href="https://github.com/aksakalli/jekyll-doc-theme">Jekyll Doc Theme</a>
        </p>
        <!-- <p class="text-muted">Place sticky footer content here.</p> -->
    </div>
</footer>

    <script>
  var baseurl = '/'
</script>
<script src="https://code.jquery.com/jquery-1.12.4.min.js"></script>
<script src="/assets/js/bootstrap.min.js "></script>
<script src="/assets/js/typeahead.bundle.min.js "></script>

<script src="/assets/js/main.js "></script>

</body>

</html>
