<!DOCTYPE html>
<html lang="en-US">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <link rel="stylesheet" href="/assets/css/main.css">
    <link rel="stylesheet" href="/assets/css/font-awesome.min.css">

    <link rel="shortcut icon" href="/favicon.ico?1">
    <!-- Begin Jekyll SEO tag v2.8.0 -->
<title>scarHRD (Homologous Recombination Deficiency) | Jake Conway</title>
<meta name="generator" content="Jekyll v3.9.3" />
<meta property="og:title" content="scarHRD (Homologous Recombination Deficiency)" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="To run scarHRD we can use the allele specific copy number calls from either FACETS or ABSOLUTE. For this example, we will be using the FACETS output." />
<meta property="og:description" content="To run scarHRD we can use the allele specific copy number calls from either FACETS or ABSOLUTE. For this example, we will be using the FACETS output." />
<link rel="canonical" href="http://localhost:4000/docs/scarHRD/" />
<meta property="og:url" content="http://localhost:4000/docs/scarHRD/" />
<meta property="og:site_name" content="Jake Conway" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2023-06-04T23:07:34+02:00" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="scarHRD (Homologous Recombination Deficiency)" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"WebPage","dateModified":"2023-06-04T23:07:34+02:00","datePublished":"2023-06-04T23:07:34+02:00","description":"To run scarHRD we can use the allele specific copy number calls from either FACETS or ABSOLUTE. For this example, we will be using the FACETS output.","headline":"scarHRD (Homologous Recombination Deficiency)","url":"http://localhost:4000/docs/scarHRD/"}</script>
<!-- End Jekyll SEO tag -->


    <link rel="alternate" type="application/rss+xml" title="Jake Conway" href="http://localhost:4000/feed.xml" />
</head>


<body>

    <nav class="navbar navbar-default navbar-fixed-top">
    <div class="container navbar-container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
            <a class="navbar-brand" href="//">
                <span><img src="/assets/img/logonav.png" alt="Logo"></span> Jake Conway
            </a>
        </div>
        <div id="navbar" class="collapse navbar-collapse">
            <ul class="nav navbar-nav">
                <li ><a href="/CV/home/">CV</a></li>
                <li  class="active" ><a href="/docs/home/">Comp Bio</a></li>
                <li ><a href="/stats/home/">Stats+ML</a></li>
                <li ><a href="/blog/2017/05/03/jekyll-doc-theme/">Blog</a></li>
            </ul>
            <div class="navbar-right">
                <form class="navbar-form navbar-left">
                    <div class="form-group has-feedback">
                        <input id="search-box" type="search" class="form-control" placeholder="Search...">
                        <i class="fa fa-search form-control-feedback"></i>
                    </div>
                </form>
                <ul class="nav navbar-nav">
                    <li><a href="https://github.com/JakeConway"><i class="fa fa-github" aria-hidden="true"></i></a></li>
                </ul>
            </div>
        </div>
    </div>
</nav>


    <div class="page-content">
        <div class="wrapper">
            <div class="container">
    <div class="row">
        <div class="col-md-4">
          <div class="panel-group" id="accordion" role="tablist" aria-multiselectable="true">
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-1" aria-expanded="false" aria-controls="collapse-1">
            Getting Started
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-1" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/home/">Welcome</a>
          
            
            
            <a class="list-group-item " href="/docs/bioconductor/">Installing Bioconductor</a>
          
            
            
            <a class="list-group-item " href="/docs/commandlinetools/">Command line tools (Docker)</a>
          
            
            
            <a class="list-group-item " href="/docs/getting_data/">Getting the data</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-2" aria-expanded="false" aria-controls="collapse-2">
            Quality Control
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-2" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewQC/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/SequencingCoverage/">Depth of Sequencing Coverage</a>
          
            
            
            <a class="list-group-item " href="/docs/CrossSampleContamination/">Cross Sample Contamination</a>
          
            
            
            <a class="list-group-item " href="/docs/TumorInNormal/">Tumor-in-normal</a>
          
            
            
            <a class="list-group-item " href="/docs/purity/">Tumor Purity</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-3" aria-expanded="false" aria-controls="collapse-3">
            Somatic Mutation Calling
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-3" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewSomaticMutations/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/MuTect/">MuTect</a>
          
            
            
            <a class="list-group-item " href="/docs/Strelka/">Strelka</a>
          
            
            
            <a class="list-group-item " href="/docs/MuTect2/">MuTect2</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-4" aria-expanded="false" aria-controls="collapse-4">
            Allelic Copy Number Calling
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-4" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewAllelicCNA/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/FACETS/">FACETS</a>
          
            
            
            <a class="list-group-item " href="/docs/GATKACNV/">GATK ACNV</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-5" aria-expanded="false" aria-controls="collapse-5">
            Mutational Signatures
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse" id="collapse-5" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewMutationalSignatures/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/SomaticSignatures/">SomaticSignatures (NMF)</a>
          
            
            
            <a class="list-group-item " href="/docs/deconstructSigs/">deconstructSigs (NNLS)</a>
          
            
            
            <a class="list-group-item " href="/docs/SigMa/">SigMa (Likelihood)</a>
          
            
            
            <a class="list-group-item active" href="/docs/scarHRD/">scarHRD (Homologous Recombination Deficiency)</a>
          
            
            
            <a class="list-group-item " href="/docs/MSIsensor/">MSIsensor (Microsatellite Instability)</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-6" aria-expanded="false" aria-controls="collapse-6">
            Mutational Significance Analysis
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-6" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewSMG/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/MutSigCV/">MutSigCV</a>
          
            
            
            <a class="list-group-item " href="/docs/OncodriveFML/">OncodriveFML</a>
          
            
            
            <a class="list-group-item " href="/docs/MutPanning/">MutPanning</a>
          
            
            
            <a class="list-group-item " href="/docs/BrownsMethod/">Brown's Method (for combining p-values)</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-7" aria-expanded="false" aria-controls="collapse-7">
            Tumor Phylogenetics
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-7" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewPhylo/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/CCF/">Calculate Mutation Cancer Cell Fractions (CCFs)</a>
          
            
            
            <a class="list-group-item " href="/docs/PhylogicNDT/">PhylogicNDT</a>
          
            
            
            <a class="list-group-item " href="/docs/PyClone/">PyClone</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-8" aria-expanded="false" aria-controls="collapse-8">
            Gene Set Analysis
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-8" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewGeneSets/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/GSEA/">GSEA</a>
          
            
            
            <a class="list-group-item " href="/docs/ssGSEA/">ssGSEA</a>
          
            
            
            <a class="list-group-item " href="/docs/PathwayOverrepresentation/">Pathway Overrepresentation</a>
          
        </div>
      </div>
    </div>
  
    <div class="panel panel-default">
      <div class="panel-heading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#collapse-9" aria-expanded="false" aria-controls="collapse-9">
            Differential Expression
          </a>
        </h4>
      </div>
      
      
      <div class="panel-collapse collapse" id="collapse-9" role="tabpanel" aria-label="Side Navigation">
        <div class="list-group">
          
            
            
            <a class="list-group-item " href="/docs/OverviewDifferentialExpression/">Overview</a>
          
            
            
            <a class="list-group-item " href="/docs/edgeR/">edgeR</a>
          
            
            
            <a class="list-group-item " href="/docs/DESeq2/">DESeq2</a>
          
            
            
            <a class="list-group-item " href="/docs/IHW/">Independent Hypothesis Weighting (IHW)</a>
          
        </div>
      </div>
    </div>
  
</div>

        </div>

        <div class="col-md-8">
            <h1>scarHRD (Homologous Recombination Deficiency)</h1>
            <div id="markdown-content-container"><p>To run scarHRD we can use the allele specific copy number calls from either FACETS or ABSOLUTE. For this example, we will be using the FACETS output.</p>

<h2 id="generating-input-file">Generating input file</h2>
<p>To generate the input file for scarHRD we need the allele specific copy number calls from the FACETS cncf table, and the tumor ploidy value.</p>

<p>If you’re running scarHRD in the same environment as the FACETS run:</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span><span class="w">

</span><span class="n">seg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">SampleID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"SAMPLE_ID"</span><span class="p">,</span><span class="w">
                  </span><span class="n">Chromosome</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">fit</span><span class="o">$</span><span class="n">cncf</span><span class="o">$</span><span class="n">chrom</span><span class="p">,</span><span class="w">
                  </span><span class="n">Start_position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">fit</span><span class="o">$</span><span class="n">cncf</span><span class="o">$</span><span class="n">start</span><span class="p">,</span><span class="w">
                  </span><span class="n">End_position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">fit</span><span class="o">$</span><span class="n">cncf</span><span class="o">$</span><span class="n">end</span><span class="p">,</span><span class="w">
                  </span><span class="n">total_cn</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">fit</span><span class="o">$</span><span class="n">cncf</span><span class="o">$</span><span class="n">tcn.em</span><span class="p">,</span><span class="w">
                  </span><span class="n">A_cn</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">fit</span><span class="o">$</span><span class="n">cncf</span><span class="o">$</span><span class="n">tcn.em</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">fit</span><span class="o">$</span><span class="n">cncf</span><span class="o">$</span><span class="n">lcn.em</span><span class="p">,</span><span class="w">
                  </span><span class="n">B_cn</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">fit</span><span class="o">$</span><span class="n">cncf</span><span class="o">$</span><span class="n">lcn.em</span><span class="p">,</span><span class="w">
                  </span><span class="n">ploidy</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">fit</span><span class="o">$</span><span class="n">ploidy</span><span class="p">)</span><span class="w">

</span><span class="c1"># remove sex chromosomes</span><span class="w">
</span><span class="n">seg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">seg</span><span class="p">[</span><span class="n">which</span><span class="p">(</span><span class="n">seg</span><span class="o">$</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="m">23</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="w">

</span><span class="c1"># format chromosome column</span><span class="w">
</span><span class="n">seg</span><span class="o">$</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">seg</span><span class="o">$</span><span class="n">Chromosome</span><span class="p">)</span><span class="w">
</span><span class="n">seg</span><span class="o">$</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s2">"chr"</span><span class="p">,</span><span class="w"> </span><span class="n">seg</span><span class="o">$</span><span class="n">Chromosome</span><span class="p">)</span><span class="w">

</span><span class="c1"># write data.table to file because scarHRD takes file path as input</span><span class="w">
</span><span class="n">write.table</span><span class="p">(</span><span class="n">seg</span><span class="p">,</span><span class="w"> </span><span class="s2">"~/Downloads/scarHRD_input.txt"</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'\t'</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>If you’re running scarHRD on the files we saved from the FACETS output:</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span><span class="w">

</span><span class="n">summary.output</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fread</span><span class="p">(</span><span class="s2">"~/Downloads/facets_summary.txt"</span><span class="p">)</span><span class="w">
</span><span class="n">seg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fread</span><span class="p">(</span><span class="s2">"~/Downloads/cncf_table.txt"</span><span class="p">)</span><span class="w">

</span><span class="c1"># make column for major copy number</span><span class="w">
</span><span class="n">seg</span><span class="o">$</span><span class="n">mcn</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">seg</span><span class="o">$</span><span class="n">tcn.em</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">seg</span><span class="o">$</span><span class="n">lcn.em</span><span class="w">

</span><span class="c1"># only keep necessary columns</span><span class="w">
</span><span class="n">seg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">seg</span><span class="p">[</span><span class="w"> </span><span class="p">,</span><span class="nf">c</span><span class="p">(</span><span class="s2">"sample_id"</span><span class="p">,</span><span class="w"> </span><span class="s2">"chrom"</span><span class="p">,</span><span class="w"> </span><span class="s2">"start"</span><span class="p">,</span><span class="w"> </span><span class="s2">"end"</span><span class="p">,</span><span class="w"> </span><span class="s2">"tcn.em"</span><span class="p">,</span><span class="w"> </span><span class="s2">"mcn"</span><span class="p">,</span><span class="w"> </span><span class="s2">"lcn.em"</span><span class="p">)]</span><span class="w">

</span><span class="c1"># add ploidy information</span><span class="w">
</span><span class="n">seg</span><span class="o">$</span><span class="n">ploidy</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">summary.output</span><span class="p">[</span><span class="n">match</span><span class="p">(</span><span class="n">seg</span><span class="o">$</span><span class="n">sample_id</span><span class="p">,</span><span class="w"> </span><span class="n">summary.output</span><span class="o">$</span><span class="n">sample_id</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="o">$</span><span class="n">ploidy</span><span class="w">

</span><span class="c1"># rename columns </span><span class="w">
</span><span class="nf">names</span><span class="p">(</span><span class="n">seg</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"SampleID"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Chromosome"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Start_position"</span><span class="p">,</span><span class="w"> </span><span class="s2">"End_position"</span><span class="p">,</span><span class="w">
                </span><span class="s2">"total_cn"</span><span class="p">,</span><span class="w"> </span><span class="s2">"A_cn"</span><span class="p">,</span><span class="w"> </span><span class="s2">"B_cn"</span><span class="p">,</span><span class="w"> </span><span class="s2">"ploidy"</span><span class="p">)</span><span class="w">

</span><span class="c1"># remove sex chromosomes</span><span class="w">
</span><span class="n">seg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">seg</span><span class="p">[</span><span class="n">which</span><span class="p">(</span><span class="n">seg</span><span class="o">$</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="m">23</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="w">

</span><span class="c1"># format chromosome column</span><span class="w">
</span><span class="n">seg</span><span class="o">$</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">seg</span><span class="o">$</span><span class="n">Chromosome</span><span class="p">)</span><span class="w">
</span><span class="n">seg</span><span class="o">$</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s2">"chr"</span><span class="p">,</span><span class="w"> </span><span class="n">seg</span><span class="o">$</span><span class="n">Chromosome</span><span class="p">)</span><span class="w">

</span><span class="c1"># write data.table to file because scarHRD takes file path as input</span><span class="w">
</span><span class="n">write.table</span><span class="p">(</span><span class="n">seg</span><span class="p">,</span><span class="w"> </span><span class="s2">"~/Downloads/scarHRD_input.txt"</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'\t'</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<h2 id="running-scarhrd">Running scarHRD</h2>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">scarHRD</span><span class="p">)</span><span class="w">

</span><span class="n">scarHRD</span><span class="o">::</span><span class="n">scar_score</span><span class="p">(</span><span class="s2">"~/Downloads/scarHRD_input.txt"</span><span class="p">,</span><span class="w">
                    </span><span class="n">seqz</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
                    </span><span class="n">reference</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grch37"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</div>
            <div style="clear:both;">
              <p class="text-center">
                <br />
                <a target="_blank" href="https://github.com/JakeConway/blob/gh-pages/_docs/Mutational Signatures/scarHRD.md" class="btn btn-default" role="button">
                  <i class="fa fa-pencil fa-lg"></i> Improve this page
                </a>
              </p>
            </div>
            <hr>
            





  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  

  
  
    <ul class="pager">
      
        
        
        <li class="previous">
          <a href="/docs/SigMa/">
            <span aria-hidden="true">&larr;</span> Previous
          </a>
        </li>
      

      
        
        
        <li class="next">
          <a href="/docs/MSIsensor/">
            Next <span aria-hidden="true">&rarr;</span>
          </a>
        </li>
      
      </ul>
    <div class="clear"></div>
    

        </div>

    </div>
</div>

        </div>
    </div>

    <footer class="footer">
    <div class="container">

        <p class="text-center">
            Jake Conway 2023 |
            Powered by <a href="https://github.com/aksakalli/jekyll-doc-theme">Jekyll Doc Theme</a>
        </p>
        <!-- <p class="text-muted">Place sticky footer content here.</p> -->
    </div>
</footer>

    <script>
  var baseurl = '/'
</script>
<script src="https://code.jquery.com/jquery-1.12.4.min.js"></script>
<script src="/assets/js/bootstrap.min.js "></script>
<script src="/assets/js/typeahead.bundle.min.js "></script>

<script src="/assets/js/main.js "></script>

</body>

</html>
